### ISSUE TITLE
buff/cache RAM not emptied

### ISSUE THREAD

**Issue by alxsimon** [bwa-mem2/bwa-mem2#100]
Hi, I noticed on my machine that when running bwa-mem2 2.0 (both the version on bioconda and the last commit of this repo) the buff/cache part of the RAM is not emptied.

This ended up posing problem after a few runs as the cache was full and bwa-mem2 runs resulted in segmentation faults.

I had to manually empty the cache to start new runs.

Under Ubuntu server 18.04.

Alexis

**Comment by mp15** [comment]
If you mean the buff/cache you can clear by doing `echo 3 | sudo tee /proc/sys/vm/drop_caches` then isn't this a kernel tuning issue? If so I think this issue is out of scope for bwa-mem2?

**Comment by alxsimon** [comment]
Sorry I honestly don't have enough knowledge about low level programming to know if it is more a kernel or software issue.
As this only happened with bwa-mem2, I thought perhaps this was linked to the software. It is fine if it isn't and the issue can be closed.
Thanks

**Comment by alxsimon** [comment]
In fact the issue is not that cache is used (which seems to be a kernel thing) but more that when RAM is full of cache, bwa-mem2 cannot run any more as it was unable to free cached memory. I ended up with segmentation faults.

**Comment by jamesluke** [comment]
Were you talking about the `buff/cache` column from running `free -m`? If so, it's possible you've been deceived by what is described here: https://www.linuxatemyram.com/

(Briefly stated, the memory in `buff/cache` is being used by the kernel for disk caching, but the kernel will drop it/repurpose it for any program that wants memory. (With a handful of exceptions for dirty pages etc.))

**Comment by alxsimon** [comment]
Yes I was talking about buff/cache which I now understand.
I initially opened this issue because at the time bwa-mem runs where dropping with seg faults quite randomly. The only thing I could think of was some issue with memory (as runs in parallel were fine if I ran them alone).
Or maybe there was an issue with the Snakemake scheduler.

Anyway, this issue can be closed I think as I did not encounter this issue afterwards.
